neuron_morphology/transforms/tilt_correction/compute_tilt_correction.py

```diff
@@ -77,41 +77,64 @@ def get_tilt_correction(morphology: Morphology,
     return tilt_angle
 
 
-def load_ccf_data(ccf_path):
-    vi = h5py.File(ccf_path, 'r')
-
-    # initialize
-    path_lookup_table = vi['view lookup'][:][np.where(vi['view lookup'][:] > -1)]
-    paths = vi['paths'][:]
-    vi.close()
-
-    voxel_idxs = []
-    path_id_from_voxel_idx = {}
-
-    for path_id in path_lookup_table:
-        path = paths[path_id, :]
-        path = path[path > 0]
-        for voxel in path:
-            path_id_from_voxel_idx[voxel] = [path_id]
-        voxel_idxs.extend(path)
-
-    return {'paths': paths,
-            'path_id_from_voxel_idx': path_id_from_voxel_idx,
-            'voxel_idxs': voxel_idxs}
+def find_closest_path(soma_voxel: List[int],
+                      ccf_path: Union[str, IO],
+                      n_sublists: int = 2):
+    '''
+        Finds closest path to soma_voxel
 
+        Parameters
+        ----------
+        soma_voxel: List containing soma voxel ccf coordinates
+        ccf_path: str, FilePath, or File-like object openable by H5PY
+        n_sublists: will seperate path_ids into n sublists to load in
+                    at a time. Higher values decrease memory usage but
+                    increase processing time. n = 1 uses about 16GB
 
-def find_closest_path_id(soma_voxel: Dict[str, int],
-                         path_id_from_voxel_idx: Dict[int, int],
-                         voxel_idxs: List[int]):
+        Returns
+        -------
+        closest_path: array of voxel coordinates of the closest streamline
+    '''
+
+    closest_path_id = None
+    min_distance = np.inf
+    with h5py.File(ccf_path, 'r') as ccf_data:
+        path_ids = np.arange(ccf_data['paths'].shape[0])
+        sublists = np.array_split(path_ids, n_sublists)
+        for sublist in sublists:
+            path_id_from_voxel_idx = {}
+            voxel_idxs = []
+            paths = ccf_data['paths'][sublist, :]
+            for i, path_id in enumerate(sublist):
+                # path = ccf_data['paths'][path_id, :]
+                path = paths[i, :]
+                path = path[path > 0]
+                for voxel_idx in path:
```

Comment:
here you're iterating over all the voxels in a path to make a look up table so that when you find the closest voxel in a block you can figure out which path it came from. Is this faster than just checking the min distance per path?

Reply:
You were correct, I was thinking of loading and testing as one thing, so either load all and test all at once or load each path individual and test once. But loading all (or 1/n) and testing each path individually is quite fast, and I think it'll reduce the memory even more! Thanks!